tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +115,119 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
+                    // this uses some underlying SqlClient code that I've copied to parse the identifier
+                    quotedName = VendoredSqlHelpers.ParseAndQuoteIdentifier(procName, isUdtTypeName: false);
+                    if (string.IsNullOrEmpty(quotedName))
+                    {
+                        // if we can't parse the identifier, return false
+                        return false;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Failed to parse/quote stored procedure, no DBM data will be propagated");
+                    return false;
+                }
+
+                // Build parameter list for EXEC statement
+                var paramList = new StringBuilder();
+                if (command.Parameters != null)
+                {
+                    foreach (DbParameter? param in command.Parameters)
+                    {
+                        if (param == null)
+                        {
+                            continue;
+                        }
+
+                        // If we have any parameters that are not Input we can't do this safely as T-SQL vs RPC calls don't honor the parameters
+                        if (param.Direction != ParameterDirection.Input)
+                        {
+                            // NOTE: we shouldn't hit this as we check above
+                            paramList.Clear();
+                            return false;
+                        }
+
+                        if (paramList.Length > 0)
+                        {
+                            // e.g. @Input=@Input, @Input2=@Input2
+                            paramList.Append(", ");
+                        }
+
+                        paramList.Append(param.ParameterName).Append('=').Append(param.ParameterName);
+                    }
+                }
+
+                // Change command type to Text, this allows us to use the EXEC statement
+                // This changes how the SQL command is executed, but since we are only supporting INPUT parameters we should be fine
+                command.CommandType = CommandType.Text;
+
+                // Create EXEC statement with parameters
+                // NOTE: EXECUTE is the exact same as EXEC, I chose EXEC arbitrarily
+                if (paramList.Length > 0)
+                {
+                    command.CommandText = $"EXEC {quotedName} {paramList} {propagationComment}";
+                }
+                else
+                {
+                    command.CommandText = $"EXEC {quotedName} {propagationComment}";
+                }
+
+                if (Log.IsEnabled(LogEventLevel.Debug))
+                {
+                    // Log the command text for debugging purposes
+                    Log.Debug("Executing stored procedure with command text: {CommandText}", command.CommandText);
+                }
```

Comment:
nit: The `Log.IsEnabled(LogEventLevel.Debug)` isn't necessary, `Log.Debug` contains this exact code. It's only useful if the values you pass to the log statement allocate or do a lot of work, which this does not
```suggestion
                // Log the command text for debugging purposes
                Log.Debug("Executing stored procedure with command text: {CommandText}", command.CommandText);
```